packages/react-vapor/src/components/datePicker/DatePickerDropdown.tsx

```diff
@@ -169,15 +158,15 @@ export class DatePickerDropdown extends React.Component<IDatePickerDropdownProps
                             </label>
                         </span>
                         <span className="dropdown-toggle-arrow"></span>
-                    </span>
+                    </button>
                     <div className={menuClasses}>{this.getDatePickerBox()}</div>
                 </div>
             </div>
         );
     }
 
-    private handleClick = () => {
-        if (this.props.onClick) {
+    private handleClick = (disabled: boolean) => {

```

Comment:
You can access `this.props.readonly` from here, so no need to pass `disabled` as parameter.